### PR TITLE
[MODDATAIMP-900] Move and improve assemble endpoint

### PR DESCRIPTION
**Pull request by ncovercash**
# [Jira MODDATAIMP-900](https://issues.folio.org/browse/MODDATAIMP-900)

## Purpose

Assembling a S3 file is analogous to uploading a file to mod-DI in the pre-S3 interface.  Therefore, we should do the same actions (marking the `FileDefinition` as uploaded, with a source path, etc)...

## Approach

To get the upload and file definition information, we moved the `assembleStorageFile` endpoint under `/uploadDefinitions/.../files/...`; after this, it was only a matter of calling the existing `afterFileSave` method.

## :sparkles: Bonus :sparkles:

- When checking coverage of S3 methods, I noticed the `downloadUrl` endpoint was missing tests, so I:
  - Added `404` response for when the information is not found, rather than a `500`.
  - Added full test suite with positive and negative cases
- Removed unnecessary invocation of Mockito in upload URL test suite

# Test coverage: 💯

+ some extra due to improvements to added download/upload url cases

## Notes

- ~~Commits through `72e386b` are part of #279 until that one gets merged in, since that PR contains some changes for WireMock I wanted to potentially use here.~~
- The try/catch around the main `DataImportImpl` logic is redundant as all potentially exceptional logic is handled inside futures, which have their own failure path.  Did the same update to download/upload URL endpoints, as I was looking at all S3 stuff.

### CODE REVIEW

**Review by okolawole-ebsco — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -129,6 +120,20 @@
...
+          "methods": [
+            "POST"
+          ],
+          "pathPattern": "/data-import/uploadDefinitions/{uploadDefinitionId}/files/{fileId}/assembleStorageFile",
+          "permissionsRequired": [
+            "data-import.assembleStorageFile.post"
+          ],
+          "modulePermissions": [
+            "configuration.entries.collection.get",
+            "change-manager.jobexecutions.post",
```
Comment (RuslanLavrov) [inline descriptors/ModuleDescriptor-template.json:133]: could you say please do we need to provide this permission (necessary for jobExecution creation) for this endpoint?

Comment (ncovercash) [reply descriptors/ModuleDescriptor-template.json:133]: good catch, thanks!

Comment (ncovercash) [reply descriptors/ModuleDescriptor-template.json:133]: We don't need the `post` one, good catch, thanks!

Comment (RuslanLavrov) [reply descriptors/ModuleDescriptor-template.json]: understood, thanks! "configuration.entries.collection.get" seems is not needed as well, right?

Comment (ncovercash) [reply descriptors/ModuleDescriptor-template.json]: @RuslanLavrov fixed!

**Review by RuslanLavrov — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (RuslanLavrov) [reply descriptors/ModuleDescriptor-template.json]: thanks!

**Pull request merged**